src/rShare.ts

```diff
@@ -0,0 +1,322 @@
+import * as vscode from 'vscode';
+import * as vsls from 'vsls';
+import * as fs from 'fs-extra';
+
+import { enableSessionWatcher, extensionContext } from './extension';
+import { attachActiveGuest, browserDisposables, initGuest, removeGuestFiles } from './rShareSession';
+import { initTreeView, rLiveShareProvider, ToggleNode } from './rShareTree';
+import { Commands, Callback, onRequest, request } from './rShareCommands';
+import { HelpFile } from './rHelp';
+
+/// LiveShare
+export let rHostService: HostService = undefined;
+export let rGuestService: GuestService = undefined;
+export let liveSession: vsls.LiveShare;
+export let isGuestSession: boolean;
+export let _sessionStatusBarItem: vscode.StatusBarItem;
+const disposables: vscode.Disposable[] = [];
+
+// service vars
+export const ShareProviderName = 'vscode-r';
+export let service: vsls.SharedServiceProxy | vsls.SharedService | null = undefined;
+
+// random number to fake a UUID for differentiating between
+// host calls and guest calls (specifically for the workspace
+// viewer 'View' function)
+export const UUID = Math.floor(Math.random() * Date.now());
+
+/// state-tracking bools
```

Comment:
To make the code of these helper functions easier to read, I'd suggest replacing the patterns
``` ts
// replace
if(CONDITION){
  return true;
} else{
  return false;
}

// with
return !!CONDITION;
```
``` ts
//replace
return A === B ? true : false;

// with
return A === B;
```
``` ts
// replace
if(CONDITION === true){}

// with
if(CONDITION){}
```